README
NAME
    Task::Curses::UI - Install the Curses::UI modules.

SYNOPSIS
     cpanm Task::Curses::UI

SEE ALSO
    Curses::UI
        A curses based OO user interface framework

    Curses::UI::AnyEvent
        Sub-class of Curses::UI for AnyEvent

    Curses::UI::Buttonbox
        Create and manipulate button widgets

    Curses::UI::Calendar
        Create and manipulate calendar widgets

    Curses::UI::Char
        Create and manipulate character widgets.

    Curses::UI::Checkbox
        Create and manipulate checkbox widgets

    Curses::UI::Color
        Color support module

    Curses::UI::Common
        Common methods for Curses::UI

    Curses::UI::Container
        Create and manipulate container widgets

    Curses::UI::DelimitedTextViewer
        Displays delimited files as fixed width.

    Curses::UI::Dialog::Basic
        Create and manipulate basic dialogs

    Curses::UI::Dialog::Calendar
        Create and manipulate calendar dialogs

    Curses::UI::Dialog::Dirbrowser
        Create and manipulate filebrowser dialogs

    Curses::UI::Dialog::Error
        Create and manipulate error dialogs

    Curses::UI::Dialog::Filebrowser
        Create and manipulate filebrowser dialogs

    Curses::UI::Dialog::Progress
        Create and manipulate progress dialogs

    Curses::UI::Dialog::Question
        Pose a simple question to the user

    Curses::UI::Dialog::Status
        Create and manipulate status dialogs

    Curses::UI::Grid
        Create and manipulate data in grid model

    Curses::UI::Grid::Cell
        Create and manipulate cell in grid model.

    Curses::UI::Grid::Row
        Create and manipulate row in grid model.

    Curses::UI::Label
        Create and manipulate label widgets

    Curses::UI::Listbox
        Create and manipulate listbox widgets

    Curses::UI::Menubar
        Create and manipulate menubar widgets

    Curses::UI::Notebook
        Create and manipulate notebook widgets.

    Curses::UI::Mousehandler::GPM
        Curses::UI GPM Bindings

    Curses::UI::Number
        Create and manipulate number widgets.

    Curses::UI::PasswordEntry
        Create and manipulate passwordentry widgets

    Curses::UI::POE
        A subclass makes Curses::UI POE Friendly.

    Curses::UI::Popupmenu
        Create and manipulate popupbox widgets

    Curses::UI::Progressbar
        Create and manipulate progressbar widgets

    Curses::UI::Radiobuttonbox
        Create and manipulate radiobuttonbox widgets

    Curses::UI::Searchable
        Add 'less'-like search abilities to a widget

    Curses::UI::TextEditor
        Create and manipulate texteditor widgets

    Curses::UI::TextEntry
        Create and manipulate textentry widgets

    Curses::UI::TextViewer
        Create and manipulate textviewer widgets

    Curses::UI::Time
        Create and manipulate time widgets.

    Curses::UI::Tutorial
        Tutorial for the Curses::UI framework

    Curses::UI::Volume
        Create and manipulate volume widgets.

    Curses::UI::Widget
        The base class for all widgets

    Curses::UI::Window
        Create and manipulate Window widgets

REPOSITORY
    <https://github.com/michal-josef-spacek/Task-Curses-UI>

AUTHOR
    Michal Josef Špaček <mailto:skim@cpan.org>

    <http://skim.cz>

LICENSE AND COPYRIGHT
    © 2015-2023 Michal Josef Špaček

    Artistic License

    BSD 2-Clause License

VERSION
    0.07



Changes
0.07

0.06 2023-12-06T12:27:19+01:00
 - Move bugtracker to github.
 - Update Curses::UI::Char to 0.02 version.
 - Update Curses::UI::Volume to 0.04 version.

0.05 2023-06-26T09:44:44+02:00
 - Add Curses::UI::Char.
 - Update Curses::UI::Number to 0.07.
 - Update Curses::UI::Time to 0.06 version.
 - Update Curses::UI::Volume to 0.03 version.
 - Update Module::Install to 1.21 version.
 - Update copyright years.

0.04 2020-04-05T15:17:10+02:00
 - Add Curses::UI::AnyEvent.
 - Add missing LICENSE file.
 - Fix Makefile.PL fails with no '.' in @INC.
 - Improve LICENSE AND COPYRIGHT section in doc.
 - Update Module::Install to 1.19 version.
 - Update author github username.
 - Update author name.
 - Update copyright years.

0.03 2015-11-05T11:58:07+01:00
 - Update Curses::UI::Number to 0.06 version.
 - Update Curses::UI::Time to 0.05 version.
 - Update Curses::UI::Volume to 0.02 version.

0.02 2015-10-12T21:01:53+02:00
 - Add modules to SEE ALSO section in doc.
 - Fix dependency for Curses::UI::Grid (Thanks shalk).
 - Move author tests to xt/ directory.
 - Update Module::Install to 1.16 version.

0.01 2015-01-21T07:47:27+01:00
 - First version


UI.pm
package Task::Curses::UI;

use strict;
use warnings;

our $VERSION = 0.07;

1;

__END__

=pod

=encoding utf8

=head1 NAME

Task::Curses::UI - Install the Curses::UI modules.

=head1 SYNOPSIS

 cpanm Task::Curses::UI

=head1 SEE ALSO

=over

=item L<Curses::UI>

A curses based OO user interface framework

=item L<Curses::UI::AnyEvent>

Sub-class of Curses::UI for AnyEvent

=item L<Curses::UI::Buttonbox>

Create and manipulate button widgets

=item L<Curses::UI::Calendar>

Create and manipulate calendar widgets

=item L<Curses::UI::Char>

Create and manipulate character widgets.

=item L<Curses::UI::Checkbox>

Create and manipulate checkbox widgets

=item L<Curses::UI::Color>

Color support module

=item L<Curses::UI::Common>

Common methods for Curses::UI

=item L<Curses::UI::Container>

Create and manipulate container widgets

=item L<Curses::UI::DelimitedTextViewer>

Displays delimited files as fixed width.

=item L<Curses::UI::Dialog::Basic>

Create and manipulate basic dialogs

=item L<Curses::UI::Dialog::Calendar>

Create and manipulate calendar dialogs

=item L<Curses::UI::Dialog::Dirbrowser>

Create and manipulate filebrowser dialogs

=item L<Curses::UI::Dialog::Error>

Create and manipulate error dialogs

=item L<Curses::UI::Dialog::Filebrowser>

Create and manipulate filebrowser dialogs

=item L<Curses::UI::Dialog::Progress>

Create and manipulate progress dialogs

=item L<Curses::UI::Dialog::Question>

Pose a simple question to the user

=item L<Curses::UI::Dialog::Status>

Create and manipulate status dialogs

=item L<Curses::UI::Grid>

Create and manipulate data in grid model

=item L<Curses::UI::Grid::Cell>

Create and manipulate cell in grid model.

=item L<Curses::UI::Grid::Row>

Create and manipulate row in grid model.

=item L<Curses::UI::Label>

Create and manipulate label widgets

=item L<Curses::UI::Listbox>

Create and manipulate listbox widgets

=item L<Curses::UI::Menubar>

Create and manipulate menubar widgets

=item L<Curses::UI::Notebook>

Create and manipulate notebook widgets.

=item L<Curses::UI::Mousehandler::GPM>

Curses::UI GPM Bindings

=item L<Curses::UI::Number>

Create and manipulate number widgets.

=item L<Curses::UI::PasswordEntry>

Create and manipulate passwordentry widgets

=item L<Curses::UI::POE>

A subclass makes Curses::UI POE Friendly.

=item L<Curses::UI::Popupmenu>

Create and manipulate popupbox widgets

=item L<Curses::UI::Progressbar>

Create and manipulate progressbar widgets

=item L<Curses::UI::Radiobuttonbox>

Create and manipulate radiobuttonbox widgets

=item L<Curses::UI::Searchable>

Add 'less'-like search abilities to a widget

=item L<Curses::UI::TextEditor>

Create and manipulate texteditor widgets

=item L<Curses::UI::TextEntry>

Create and manipulate textentry widgets

=item L<Curses::UI::TextViewer>

Create and manipulate textviewer widgets

=item L<Curses::UI::Time>

Create and manipulate time widgets.

=item L<Curses::UI::Tutorial>

Tutorial for the Curses::UI framework

=item L<Curses::UI::Volume>

Create and manipulate volume widgets.

=item L<Curses::UI::Widget>

The base class for all widgets

=item L<Curses::UI::Window>

Create and manipulate Window widgets

=back

=head1 REPOSITORY

L<https://github.com/michal-josef-spacek/Task-Curses-UI>

=head1 AUTHOR

Michal Josef Špaček L<mailto:skim@cpan.org>

L<http://skim.cz>

=head1 LICENSE AND COPYRIGHT

© 2015-2023 Michal Josef Špaček

Artistic License

BSD 2-Clause License

=head1 VERSION

0.07

=cut


Makefile.PL
use lib '.';
use strict;
use warnings;

use inc::Module::Install;

# Definition.
abstract 'Install the Curses::UI modules.';
author 'Michal Josef Spacek <skim@cpan.org>';
author_requires 'English' => 0;
author_requires 'File::Object' => 0.08;
author_requires 'Test::More' => 0;
author_requires 'Test::NoWarnings' => 0;
author_requires 'Test::Pod' => 0;
author_requires 'Test::Pod::Coverage' => 0;
license 'bsd';
name 'Task-Curses-UI';
readme_from 'UI.pm';
recursive_author_tests('xt');
requires 'Curses::UI' => 0.9609;
requires 'Curses::UI::AnyEvent' => 0.101;
requires 'Curses::UI::Char' => 0.02;
requires 'Curses::UI::DelimitedTextViewer' => 0.10;
requires 'Curses::UI::Grid' => 0.15;
requires 'Curses::UI::Mousehandler::GPM' => 0.01;
requires 'Curses::UI::Number' => 0.07;
requires 'Curses::UI::POE' => 0.04;
requires 'Curses::UI::Time' => 0.06;
requires 'Curses::UI::Volume' => 0.04;
requires 'perl' => '5.6.0';
resources 'bugtracker' => 'https://github.com/michal-josef-spacek/Task-Curses-UI/issues';
resources 'homepage' => 'https://github.com/michal-josef-spacek/Task-Curses-UI';
resources 'repository' => 'git://github.com/michal-josef-spacek/Task-Curses-UI';
test_requires 'Test::More' => 0;
test_requires 'Test::NoWarnings' => 0;
tests_recursive;
version '0.07';

# Run.
WriteAll();
